openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/core/AbstractCoreRoutingAlgorithm.java

```diff
@@ -37,29 +40,42 @@
     int visitedCountTo1;
     int visitedCountFrom2;
     int visitedCountTo2;
-    int visitedEdgesALTCount;
 
     private CoreDijkstraFilter additionalCoreEdgeFilter;
 
     boolean inCore;
 
-    public AbstractCoreRoutingAlgorithm(Graph graph, Weighting weighting, TraversalMode tMode) {
-        super(graph, weighting, tMode);
+    protected final TurnWeighting turnWeighting;
+    protected final boolean hasTurnWeighting;
+    protected boolean approximate = false;
+
+    public AbstractCoreRoutingAlgorithm(Graph graph, Weighting weighting) {
+        super(graph, new PreparationWeighting(weighting), TraversalMode.NODE_BASED);
+
+        if (weighting instanceof TurnWeighting) {
+            turnWeighting = (TurnWeighting) weighting;
+            hasTurnWeighting = true;
+        }
+        else {
+            turnWeighting = null;
```

Comment:
Both `turnWeighting` and `hasTurnWeighting` have been originally declared as `final`. As such they needed to be initialized with `null` and `false`, respectively. Because it is not critical for them to be `final` I guess we can omit this for the sake of simpler code.